src/swagger/OpenAPITypesGuard.ts

```diff
@@ -56,18 +56,30 @@ export class OpenAPITypesGuard {
     }
 
     public isString(schema: SchemaType): schema is IOpenAPI3StringSchema {
-        return (schema as IOpenAPI3StringSchema)?.type === 'string';
+        return this.isStringType(schema) && !this.isDateFormat(schema) && !this.isGuidFormat(schema);
```

Comment:
it's very strange. Maybe 
(schema as IOpenAPI3StringSchema)?.type === 'string' && schema.format === undefined

Reply:
or

isString(schema) && isFormatUndefined(..